Dynamics/Delay.agda

```diff
@@ -0,0 +1,279 @@
+module Dynamics.Delay where 
+  open import Cubical.Foundations.Prelude
+  open import Cubical.Categories.Category 
+  open import Cubical.Categories.Functor 
+  open import Cubical.Foundations.HLevels
+  open import Cubical.Foundations.Structure hiding(str)
+  open import Cubical.Categories.Instances.Sets
+  open import Cubical.Categories.Limits.Terminal
+  open import Cubical.Categories.Instances.FunctorAlgebras
+  open import Cubical.Foundations.Isomorphism
+  open import Cubical.Foundations.HLevels
+  open import Cubical.Foundations.Function
+  open import Cubical.Data.Nat
+  import Cubical.Data.Equality as Eq
+  open import Cubical.Data.Sum renaming (rec to ⊎rec)
+  open import Cubical.Data.Unit renaming (Unit to ⊤ )
+  open Category
+  open Functor
+  open Algebra
+  open AlgebraHom
+  open import Cubical.Data.Sigma.Properties
+  
+  private
+    variable
+      ℓ : Level
+  
+  module Basics where 
+
+    data StateF (Res Rec : Type ℓ) : Type ℓ where
```

Comment:
this is just a coproduct?

Reply:
yea... should I just use coproduct instead?